FilecoinFoundationWeb/filecoin-foundation
# [UXIT-1540] Refactor SEO title metadata

## 📝 Description

This PR includes refactor in order to include an optional SEO title for dynamic pages - events, ecosystem project, digest articles and blog posts. If left empty, it defaults to page title. For events, we add a suffix "- Event", and for ecosystem project, we add "- Ecosystem Project".

**Note**: in `createMetadata` file, I encountered some type errors for the `title`- if you have a better idea, probably there's a better workaround than using `as string`?

```
    title: overrideDefaultTitle
      ? { absolute: parsedEnrichedSEO.title **as string** }
      : (parsedEnrichedSEO.title **as string**),
```

- **Type:** Refactor

## 🛠️ Key Changes

- `meta_config` and `SeoMetadataSchema` are broken down in smaller parts

## 📌 To-Do Before Merging

- [x] Check with David 


## 🧪 How to Test

  1. open decap cms
  2. check if static pages have mandatory SEO title and dynamic page an optional SEO title
  3. delete SEO title in dynamic page
  4. check (either console or dev tools) if the SEO title matches the title (It should not be empty). For events, it should have a suffix `"- Events"`, and for ecosystem projects` "- Ecosystem Project"`. Example. 'Acasia - Ecosystem Project'
